dev/tests/integration/testsuite/Magento/Shipping/Block/Adminhtml/Order/AddToPackageTest.php

```diff
@@ -42,30 +51,43 @@ protected function setUp(): void
     }
 
     /**
-     * Loads order entity by provided order increment ID.
+     * Test that Packaging popup renders
      *
-     * @param string $incrementId
-     * @return OrderInterface
+     * @magentoDataFixture Magento/Shipping/_files/shipping_with_carrier_data.php
      */
-    private function getOrderByIncrementId(string $incrementId) : OrderInterface
+    public function testGetCommentsHtml(): void
     {
-        /** @var SearchCriteria $searchCriteria */
-        $searchCriteria = $this->objectManager->get(SearchCriteriaBuilder::class)
-            ->addFilter('increment_id', $incrementId)
-            ->create();
-
-        $items = $this->orderRepository->getList($searchCriteria)
-            ->getItems();
+        $expectedNeedle = "packaging.setItemQtyCallback(function(itemId){
+            var item = $$('[name=\"shipment[items]['+itemId+']\"]')[0],
+                itemTitle = $('order_item_' + itemId + '_title');
+            if (!itemTitle && !item) {
+                return 0;
+            }
+            if (item && !isNaN(item.value)) {
+                return item.value;
+            }
+        });";
+        $this->assertStringContainsString($expectedNeedle, $this->getHtml());
+    }
 
-        return array_pop($items);
+    /**
+     * Verify currency code on custom value field
+     *
+     * @magentoDataFixture Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php
+     */
+    public function testGetCurrencyCodeCustomValue ()
+    {
+        $expectedCurrencyCode = '<span class="customs-value-currency">
```

Comment:
This check is too unstable. Any change in formatting will break test
I Suggest extract value and check it
```
$template = '/<span class="customs-value-currency">\s*?(?<currency>[A-Za-z]+)\s*?<\/span>/';
$matches = [];
preg_match($template, $text, $matches);
$currency = $matches['currency'] ?? null;
assert...
```